### PR TITLE
TestRunner refactor round 2, keep decoupling and build test lifecycle: buildDeviceTestTask -> setUp -> run -> tearDown

### PR DESCRIPTION
**Pull request by hydraxman**
- Test runner lifecycle: buildDeviceTestTask -> setUp -> run -> tearDown
- Remove runner class dependencies for Spring annotations, create TestRunnerConfig.java to centralize the configuration.
- Inline some runner members as local var.
- Change the method access of reInstallApp
- Fix bug in ADBLogcatCollector isCrashFound() method.

### CODE REVIEW

File: common/src/main/java/com/microsoft/hydralab/common/logger/impl/ADBLogcatCollector.java [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import com.microsoft.hydralab.common.management.DeviceManager;
 import com.microsoft.hydralab.common.util.ADBOperateUtil;
 import com.microsoft.hydralab.common.util.LogUtils;
+import org.apache.commons.lang3.StringUtils;
```
Comment (dinghong2233) [inline common/src/main/java/com/microsoft/hydralab/common/logger/impl/ADBLogcatCollector.java:11]: Remove unused import please

Comment (dinghong2233) [reply common/src/main/java/com/microsoft/hydralab/common/logger/impl/ADBLogcatCollector.java:11]: My bad.

**Review by dinghong2233 — APPROVED** [review]

File: agent/src/main/java/com/microsoft/hydralab/agent/runner/TestRunner.java [hunk 2]
```diff
@@ -76,30 +137,8 @@ private void presetForMicrosoftLauncherApp(DeviceInfo deviceInfo, TestTask testT
...
-        reportLogger.info("Finally: restore state");
-        /* Close/finish resource */
-        if (reportLogger != null) {
-            reportLogger.info("Start Close/finish resource");
-            LogUtils.releaseLogger(reportLogger);
-        }
-    }
 
-    public void reInstallApp(DeviceInfo deviceInfo, TestTask testTask, Logger reportLogger) throws Exception {
+    protected void reInstallApp(DeviceInfo deviceInfo, TestTask testTask, Logger reportLogger) throws Exception {
```
Comment (zhou9584) [inline agent/src/main/java/com/microsoft/hydralab/agent/runner/TestRunner.java:141]: Maybe we should replace the implement of this method in parent class and child classes. Only espresso need to install 2 app, the rest like monkey, smart and appium just need 1 apk.
image

File: agent/src/main/java/com/microsoft/hydralab/agent/runner/TestRunner.java [hunk 3]
```diff
@@ -139,18 +143,32 @@ private void presetForMicrosoftLauncherApp(DeviceInfo deviceInfo, TestTask testT
...
+    }
 
+    protected void reInstallTestApp(DeviceInfo deviceInfo, TestTask testTask, Logger reportLogger) throws Exception {
+        if (testTask.getTestAppFile() == null) {
+            return;
+        }
+        if (StringUtils.isEmpty(testTask.getTestPkgName())) {
+            return;
+        }
+        if (!testTask.getTestAppFile().exists()) {
```
Comment (zhou9584) [inline agent/src/main/java/com/microsoft/hydralab/agent/runner/TestRunner.java:164]: Currently, the testAppFile could be .apk file (Espresso) or .jar file(Appium). So, we could not install it directly.

Comment (hydraxman) [reply agent/src/main/java/com/microsoft/hydralab/agent/runner/TestRunner.java:164]: Will this work?
image

**Review by zhou9584 — APPROVED** [review]

**Pull request merged**